### ISSUE TITLE
Provide a simple built-in way to change the file extension used for .NET assemblies in Blazor WebAssembly apps

### ISSUE THREAD

**Issue by danroth27** [dotnet/aspnetcore#21489]
We documented how you can script changing the file extension used for .NET assemblies in a Blazor WebAssembly app. But it would be nice to have a simple built-in way to do this.

[Suggestion](https://github.com/dotnet/aspnetcore/issues/5477#issuecomment-623167309) from @stavroskasidis:

> Hopefully one day this could all be handled by the framework with a flag in the csproj, e.g.
> ```xml
> <BlazorBinariesExtension>.bin</BlazorBinariesExtension>
> ```

**Comment by mihaimyh** [comment]
From what I can see the rename from .dll to .bin must be done also in `wwwroot/service-worker-assets.js` and remove or re-compress `wwwroot/service-worker-assets.js.br` and `wwwroot/service-worker-assets.js.gz`, otherwise there are integrity issues in browser's console.

**Comment by danroth27** [comment]
> From what I can see the rename from .dll to .bin must be done also in `wwwroot/service-worker-assets.js` and remove or re-compress `wwwroot/service-worker-assets.js.br` and `wwwroot/service-worker-assets.js.gz`, otherwise there are integrity issues in browser's console.

@guardrex @SteveSandersonMS Do you need to update the docs on this to account for this?

**Comment by guardrex** [comment]
The current doc does **NOT** address *service-worker-assets.js*; so if that's an issue, we'll need the doc to reflect it.

After I hear back that it should be called out, I'll open an doc issue to address it. :ear:

**Comment by SteveSandersonMS** [comment]
Yes, the same filename replacements would need to be made in the service worker assets manifest too.

**Comment by darcythomas** [comment]
I wonder if there is a risk some AV scanners may inspect files looking for dll magic strings.?! So it would be nice if there were some way of obfuscating the files. Say by a simple XOR encryption with a random key that is included in the `blazor.boot.json` Or if you want to be fancy using `CryptoJS.AES`

Thoughts?

**Comment by smartprogrammer93** [comment]
I believe this should get more love since https://github.com/dotnet/AspNetCore.Docs/issues/18551 
and cases like https://github.com/dotnet/aspnetcore/issues/21996 might happen more often with dll file ending

**Comment by guardrex** [comment]
@smartprogrammer93 asked about guidance for compression. Today, we just say that it's up to the dev to deal with recompressing *blazor.boot.json*. Should I seek to expand the doc scripts to handle compression? I'd rather **_not_** if the framework is going to get something built in. I'm 🏃 on a lot of high priority work for the next few months.

**Comment by SteveSandersonMS** [comment]
@pranavkm That'll teach you to ask more than one question. Loving the ambiguity :)
